ext/sodium/libsodium.c

```diff
@@ -387,7 +387,8 @@ static void sodium_remove_param_values_from_backtrace(zend_object *obj) {
 		ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(trace), frame) {
 			if (Z_TYPE_P(frame) == IS_ARRAY) {
 				zval *args = zend_hash_str_find(Z_ARRVAL_P(frame), "args", sizeof("args")-1);
-				if (args && Z_TYPE_P(frame) == IS_ARRAY) {
+				if (args != NULL && Z_TYPE_P(args) == IS_ARRAY &&
+                    Z_REFCOUNTED_P(args) && Z_REFCOUNT_P(args) == 1) {
```

Comment:
As the intention here is to hide arguments, this check would kinda defeat the point in the cases where it triggers. As an alternative, I'd suggest doing something like

```
zval *args = zend_hash_str_find(Z_ARRVAL_P(frame), "args", sizeof("args")-1);
zval_ptr_dtor(args);
ZVAL_EMPTY_ARRAY(args);
```

in order to overwrite with an empty array, without corrupting the original one.

Reply:
cc @jedisct1